test_runner/src/main/kotlin/ftl/domain/README.md

```diff
@@ -0,0 +1,56 @@
+# Domain
+
+Owns business logic of the application.
+
+## Design
+
+The domain layer is exposing own API as a collection of public suspendable extension functions,
+which could be called a `top-level functions`.
+Each `top-level function` have own `execution context`, 
+can produce `structured output` during the `execution`
+and can be composed of one or many low-level functions.
+This layer could be considered as a standalone library, 
+that is providing access to business logic through pure kotlin functions. 
+
+
+### Top-level function
+
+Is public and accessible from the root of the domain package.
+IS responsible to run domain logic directly or compose it using low-level domain functions, utils, or data interfaces.
+For simplification, consider a simple common interface for all top-level functions.
+
+```kotlin
+typealias UseCase<A> = suspend A.() -> Unit
+```
+
+Where `A` is a generic type that is representing the `execution context`.
+
+### Execution context
+
+The context can provide arguments for the execution and its name should reflect the related use case.
+
+### Low-level function
+
+The low-level function is useful when it comes to divide complex top-level 
+function into the composition of smaller chunks or reuse some logic in many top-level functions.
+Keeping the composition of low-level functions flat is crucial. 
+More nesting in depth, can make a code much harder to understand and maintain.
+
+## Responsibility
+
+* Contain a business logic of application
```

Comment:
Two possibilities here choose one :) 
```suggestion
* Contains the business logic of the application
```
OR
```suggestion
* Contains one business logic concern of the application per use-case
```

Reply:
I prefer the shorter one :)